src/sentry/api/endpoints/organization_integration_details.py

```diff
@@ -5,35 +5,26 @@
 import six
 from django.http import Http404
 
-from sentry.api.bases.organization import OrganizationEndpoint, OrganizationIntegrationsPermission
+from sentry.api.bases.organization import OrganizationIntegrationsPermission
+from sentry.api.bases.organization_integrations import OrganizationIntegrationBaseEndpoint
 from sentry.api.serializers import serialize
-from sentry.shared_integrations.exceptions import IntegrationError
 from sentry.models import Integration, ObjectStatus, OrganizationIntegration
+from sentry.shared_integrations.exceptions import IntegrationError
 from sentry.tasks.deletion import delete_organization_integration
 
 
-class OrganizationIntegrationDetailsEndpoint(OrganizationEndpoint):
+class OrganizationIntegrationDetailsEndpoint(OrganizationIntegrationBaseEndpoint):
     permission_classes = (OrganizationIntegrationsPermission,)
 
     def get(self, request, organization, integration_id):
-        try:
-            integration = OrganizationIntegration.objects.get(
-                integration_id=integration_id, organization=organization
-            )
-        except OrganizationIntegration.DoesNotExist:
-            raise Http404
+        integration = self.get_organization_integration(organization, integration_id)
```

Comment:
@mgaeta The variable here should be called `org_integration` to avoid confusion with `integration`.